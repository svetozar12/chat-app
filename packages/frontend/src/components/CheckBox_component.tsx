import React from "react";

export const CheckBox_component = ({
  item,
  invited,
  setInvited,
  allChecked,
}: {
  item: string;
  invited: string[];
  setInvited: React.Dispatch<React.SetStateAction<string[]>>;
  allChecked: boolean;
}) => {
  const [isChecked, setIsChecked] = React.useState(false);
  React.useEffect(() => {
    if (isChecked) {
      setInvited([...invited, item]);
      return;
    }
    const newArr = invited.filter((element) => element != item);
    setInvited(newArr);
  }, [isChecked]);

  return (
    <div
      onClick={() => setIsChecked(!isChecked)}
      className="flex add_users_checkBox"
    >
      <p style={{ fontSize: "1.3rem" }}>{item}</p>
      <input
        type="checkbox"
        checked={(allChecked && false) || isChecked}
        value={item}
      />
    </div>
  );
};
