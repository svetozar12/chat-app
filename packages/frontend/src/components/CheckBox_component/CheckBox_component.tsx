import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/css";
const Add_users_checkBox = styled.div`
  width: 95%;
  border-top: 1px solid rgba(0, 0, 0, 0.3);
  justify-content: space-between;
  cursor: pointer;
`;

const Suggested_user = styled.p`
  font-size: 1.3rem;
`;
export const CheckBox_component = ({
  item,
  invited,
  setInvited,
}: {
  item: string;
  invited: string[];
  setInvited: React.Dispatch<React.SetStateAction<string[]>>;
}) => {
  const [isChecked, setIsChecked] = React.useState(false);
  React.useEffect(() => {
    if (isChecked) {
      setInvited([...invited, item]);
      return;
    }
    const newArr = invited.filter((element) => element != item);
    setInvited(newArr);
  }, [isChecked]);

  return (
    <Add_users_checkBox
      title="checkbox"
      onClick={() => setIsChecked(!isChecked)}
      className="flex"
    >
      <Suggested_user>{item}</Suggested_user>
      <input
        className={css`
          width: 1.25rem;
          height: 1.25rem;
          margin: 0 1rem;
        `}
        type="checkbox"
        onChange={() => setIsChecked(!isChecked)}
        checked={isChecked}
        value={item}
      />
    </Add_users_checkBox>
  );
};
