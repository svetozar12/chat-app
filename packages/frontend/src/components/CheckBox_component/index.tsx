export { default } from "./CheckBox_component";
