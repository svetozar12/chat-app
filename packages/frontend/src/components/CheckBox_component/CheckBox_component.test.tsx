import { CheckBox_component } from "./CheckBox_component";
import renderer from "react-test-renderer";
import { render, cleanup, RenderResult } from "@testing-library/react";
import { ReactTestRendererJSON } from "react-test-renderer";
import "@testing-library/jest-dom";

let component: ReactTestRendererJSON | ReactTestRendererJSON[] | null;
let container: RenderResult;
const submit: any = jest.fn();

beforeEach(() => {
  component = renderer.create(<CheckBox_component item={"greg"} invited={["greg", "ivan"]} setInvited={submit} />).toJSON();

  container = render(<CheckBox_component item={"greg"} invited={["greg", "ivan"]} setInvited={submit} />);
});

afterEach(cleanup);

describe("Render connected React-redux page", () => {
  it("should create snapshot for <CheckBox_component/>", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render <CheckBox_component/>", () => {
    const renderedComponent = container.getByTitle("checkbox");
    expect(renderedComponent).toBeInTheDocument();
  });
});
