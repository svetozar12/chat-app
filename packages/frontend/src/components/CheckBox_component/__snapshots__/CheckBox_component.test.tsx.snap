// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <CheckBox_component/> 1`] = `
<div
  className="flex css-16yups9"
  onClick={[Function]}
  title="checkbox"
>
  <p
    className="css-1nu6ntq"
  >
    greg
  </p>
  <input
    checked={false}
    className="css-jgjwtu"
    onChange={[Function]}
    type="checkbox"
    value="greg"
  />
</div>
`;
